woocommerce/wc-admin
# Unbundle wp.components

Fixes https://github.com/woocommerce/woocommerce-admin/issues/5034

Bundling `wp.components` has become technical debt and this PR eliminates it from the bundle in favour of the version found on the window.

By bumping the minimum version of `Requires at least` to 5.4.0, we are sure that React Hooks are available. This is also in line with Core's minimum version requirements to support current and last two minor versions now that WP 5.6 has been released.

This PR automatically turns the Navigation feature flag off for users of WP less than WP 5.6.0 because the `Navigation` component is not available. If Gutenberg feature plugin is active and greater than 9.0.0, then Navigation feature is enabled.

This PR does the following

* Enqueue `wp.components`
* Bump minimum WP requirements
* Disable Navigation feature if WP or Gutenberg requirements aren't met

### Detailed test instructions:

1. Use WP 5.5.x.
2. Turn the option on to enable Navigation by setting `woocommerce_navigation_enabled` to "yes".
3. Disable Gutenberg feature plugin if you have it enabled.
4. Disable wc-calypso-bridge in case you were testing a branch that forces Navigation on.
5. Run `npm start`. Not doing so could carry over a feature flag config from a previous branch.
6. See Navigation not turned on.
7. Activate the Gutenberg plugin.
8. See the Navigation turned on.
9. Deacitvate Gutenberg. Poke around the app to make sure there are no regressions.
10. Update to WP 5.6
11. See the Nav turned on.
12. Poke around the app to make sure there are no regressions.
13. Repeat the steps with WP 5.4.x installed. Be sure to run through the OBW. You can also use this JN site:
* https://presidential-marmoset.jurassic.ninja/wp-admin/admin.php?page=wc-admin
* un: demo
* pw: G4pRipcXseMz

### Consequences

* 650kb reduction in bundle measured using `npm ci && npm run analyze`. This isn't as much as [previously measured](https://github.com/woocommerce/woocommerce-admin/pull/5144) but I'm not sure why.
* Adding new features available only in the latest version of Gutenberg will need to be behind a feature so we can check version compatibility.
* Until a proper [Dependency Extraction Webpack Plugin is added](https://github.com/woocommerce/woocommerce-admin/issues/5647), extensions may experience fatals due to dependencies not correctly enqueued.
* Upstream Gutenberg contribution can't be made easily available. Some strategy will be required to add just components we absolutely need updated. A strategy to easily revert this when an update becomes available on the minimum WP version will be useful too.
* We can remove SlotFill re-export workarounds https://github.com/WordPress/gutenberg/issues/27191

### Known Issues and Follow Ups

* Add a note to Feature Setting https://github.com/woocommerce/woocommerce-admin/issues/5871
* Nav "back" button issues https://github.com/woocommerce/woocommerce-admin/issues/5872
* Handle potential problems with `__experimental` prefix https://github.com/woocommerce/woocommerce-admin/issues/5873
* Add Dependency Extraction Webpack Plugin https://github.com/woocommerce/woocommerce-admin/pull/5762

### Changelog Note:

- Dev: Unbundle `wp.components` and enqueue it instead.
